main/src/ca/uwaterloo/flix/language/phase/Weeder2.scala

```diff
@@ -716,8 +716,9 @@ object Weeder2 {
       ) {
         case Some(expr) => Validation.success(expr)
         case None =>
+          // Fall silently back on Expr.Error. Parser should produce error here.
```

Comment:
We are returning a `success` because a `softFailure` would add another error message. I don't think we want that, because in the case of an syntactically incorrect program, I think it is best to output a syntax error only. A type error or resolution error is very likely to be of low quality, because the program is in a state that's very hard to reason about.

I agree that the `Validation.success(Expr.Error(...))` doesn't have a nice ring to it! But I think it is the only way to perpetuate the error, while not outputting anything to the programmer (to "fail silently", in a lack of better wording). I didn't want to make changes to `Validation` to facilitate "a failure with nothing to output".

> We don't have to do this anywhere else, why do we have to do it here?

I think we do have to do something similar elsewhere. In the Typer for instance, we should make sure not to produce an error ala "Expected Int32 but found Error". In general, any phase after the Parser should ignore error nodes as much as possible. They can't be reasoned with.

Reply:
I think something like "Fall back on Expr.Error. The Parser has reported an error here." is ok.